### PR TITLE
Removed redundant closure literals

### PR DESCRIPTION
**Pull request by tardieu**
## Description

Removed redundant closure literals

## Motivation and Context

Code cleanup

## How Has This Been Tested?

Ran swift test

## Checklist:

- [ ] I have submitted a [CLA form](https://github.com/IBM-Swift/CLA)
- [ ] If applicable, I have updated the documentation accordingly.
- [ ] If applicable, I have added tests to cover my changes.

### CODE REVIEW

File: Sources/KituraNet/IncomingSocketHandler.swift [hunk 1]
```diff
@@ -63,12 +63,8 @@ public class IncomingSocketHandler {
...
                                                          queue: IncomingSocketHandler.socketReaderQueue[Int(socket.socketfd%2)])
         
-            readerSource.setEventHandler() {
-                self.handleRead()
-            }
-            readerSource.setCancelHandler() {
-                self.handleCancel()
-            }
+            readerSource.setEventHandler(handler: self.handleRead)
+            readerSource.setCancelHandler(handler: self.handleCancel)
```
Comment (shmuelk) [inline Sources/KituraNet/IncomingSocketHandler.swift]: Does these statements cause handleRead and handleCancel to be called in the context of self? I have to admit that I don't understand how the tests would pass if that wasn't the case. I'm just trying to double check.

File: Sources/KituraNet/IncomingSocketHandler.swift [hunk 2]
```diff
@@ -145,9 +141,7 @@ public class IncomingSocketHandler {
             writerSource = DispatchSource.makeWriteSource(fileDescriptor: socket.socketfd,
                                                           queue: IncomingSocketHandler.socketWriterQueue)
             
-            writerSource!.setEventHandler() {
-                self.handleWriteHelper()
-            }
+            writerSource!.setEventHandler(handler: self.handleWriteHelper)
```
Comment (shmuelk) [inline Sources/KituraNet/IncomingSocketHandler.swift]: Same comment as above.....

File: Sources/KituraNet/IncomingSocketHandler.swift (hunk 1 above)
Comment (tardieu) [inline Sources/KituraNet/IncomingSocketHandler.swift]: I believe so. "self.handleRead" is a member reference expession just like a field access. The original code introduces a redudant closure that does nothing more than invoking the function.  I have not looked precisely at the generated code, but the extra closure does show on performance profiles so this looks like a straightforward optimization.

File: Sources/KituraNet/IncomingSocketHandler.swift (hunk 2 above)
Comment (tardieu) [inline Sources/KituraNet/IncomingSocketHandler.swift]: Same thing.

**Comment by ianpartridge** [comment]
This has conflicts now, unfortunately. @tardieu can you fix up your branch? I'd like to get this merged.

**Comment by tardieu** [comment]
I have updated the branch as requested.

**Comment by ianpartridge** [comment]
I'm a bit confused, this is showing as 29 commits with a diff of "+990 −719". Did you rebase on master?

**Comment by tardieu** [comment]
Take 2... I did a simple merge this time.

**Pull request merged**